### ISSUE TITLE
Support multi-comment ignores with `# pyright: ignore` comments

### ISSUE THREAD

**Issue by adam-grant-hendry** [microsoft/pyright#3570]
## Problem

`pylance` does not support multi-comment ignores. The typical syntax for this is:

```python
# linter1: ignoreSyntax  # linter2: ignoreSyntax  # linter3: ignoreSyntax  # etc.
```

This disables inline ignores for other linters (`flake8`, `pylint`, python's built-in `type`, etc.)

## Environment data

-   Language Server version: v2022.6.11
-   OS and version: Windows 16
-   Python version (& distribution if applicable, e.g. Anaconda): 3.8.10

## Code Snippet

Example snippet

```python
from qtpy import QtCore, QtWidgets

app = QtWidgets.QApplication([])
app.setStyle('fusion')  # type: ignore[misc, call-overload]  # mypy Issue 6910
app.setAttribute(
    QtCore.Qt.AA_DontShowIconsInMenus, True
)  # pyright: ignore  # cSpell:ignore Dont  # This fails (pyright does not honor multi-comment inline ignores)

...
```

## Repro Steps

See code example above for example

## Expected behavior

`pylance`/`pyright` honor multi-comment inline ignores

## Actual behavior

`pylance`/`pyright` do not honor multi-comment inline ignores

**Comment by adam-grant-hendry** [comment]
@erictraut Actually, it appears the problem is the comment line being moved by `black` to the new line after the closing `)`. If the multi-comment ignore is on the same line as the offending line, the ignore is honored.

**Comment by rhoit** [comment]
I'm using pyright 1.1.273

with pyright which triggers flake8 max-line-length

![image](https://user-images.githubusercontent.com/1684754/193492448-694e6618-3006-4529-9ad4-63a5a0cc3041.png)

with flake8 ignore max-line-length and pyright which ignore the multi line comment.

![image](https://user-images.githubusercontent.com/1684754/193492376-403c7a8d-e83f-4b50-b595-8d0bf1999587.png)

**Comment by erictraut** [comment]
This issue has been closed for a long time. If you have questions, please post to the discussion forum. If you would like to report a bug or file an enhancement request, please open a new issue.

You are using `# pyright: reportGeneralTypeIssues=false` incorrectly. This comment form is intended to be used at the top of a file, and it suppresses all errors within the file that are generated by the `reportGeneralTypeIssues` diagnostic check. If your intent is to suppress a diagnostic on a single line, you should use `# pyright: ignore[reportGeneralTypeIssues]`.

**Comment by rhoit** [comment]
@erictraut thanks for pointing out the incorrect usage, I just started using pyright this is my second day! 

everything is ok!

![image](https://user-images.githubusercontent.com/1684754/193493278-d92d43de-01f4-469f-830b-24447b410ff9.png)
